### DISCUSSION TITLE
Error on some Ultimate Parent Rollups

### DISCUSSION THREAD

**Discussion by fabm97** [jamessimone/apex-rollup#274] [Q&A]
Hey,

we recently discovered some of our rollups failing due to a query issue:
```
Failed to process batch for class 'RollupFullBatchRecalculator' for job id '7071v0000Ao1GcR'

caused by: System.QueryException:
SELECT Id,AdjustedCCId__c,Meter__c,Is_Valid__c,CurrencyIsoCode
                                ^
ERROR at Row:1:Column:33
No such column 'Parent_Meter__c' on entity 'Meter2GenerationUnitRelationship__c'. If you are attempting to use a custom field, be sure to append the '__c' after the custom field name. Please reference your WSDL or the describe call for the appropriate names.

Class.RollupRelationshipFieldFinder.retrieveAdditionalCalcItems: line 540, column 1
```

The field Parent_Meter__c only exists on the Lookup Object "Meter" and as I understand it should normally also only be queried from Meter__c and not the Calc Item Object Meter2GenerationUnitRelationship__c.

The error only happens on our production system and not on any test system (all systems are on 1.4.7) although the Rollup Metadata Record is the same. I'd guess this may be an error happening only with specific data constellation, as it also only occurs in 2 of 70 batches of the Rollup Recalculation.
The metadata record looks like this:
![image](https://user-images.githubusercontent.com/40338881/160383764-6164cfee-4958-4e06-8e1d-ce86cdf998b6.png)

I hope you can maybe help us figure this out, because I think this may be some general bug in Apex Rollup.
(Edit: Created as issue by accident, moved it now into discussion)

**Comment by jamessimone** [comment]
@fabm97 thanks for reporting this - I’m also ok with it being logged as an issue; even if it doesn’t happen every time, I’ve seen the error emails for this come through often enough to be convinced it’s some edge case worth tracking down. I will try to reproduce the issue.

**Reply by jamessimone** [reply]
@fabm97 sorry for the delay on this - I’m going to create an issue out of this discussion as I keep losing track of this one and I want to finally get this resolved

**Reply by fabm97** [reply]
@jamessimone Thanks for the update :) I tried to figure out a pattern in this error in the meantime but since the 3rd April the error is gone on our Production Org without any changes from my side... So sadly I also have no further clues on why it happened and currently just hope it stays working as of now

**Reply by jamessimone** [reply]
That’s really good to know - did you upgrade Apex Rollup at any point prior to the error disappearing? I hadn’t considered that I might have accidentally fixed this - which I usually try to indicate with tests - while doing some other work surrounding this functionality. I will still try to produce tests that _prove_ this is no longer an issue.

**Reply by fabm97** [reply]
@jamessimone I installed the latest update available on 23rd of March (1.4.7) to see if the issue may be fixed by that. The error then continued to come up until April 3rd.

**Reply by fabm97** [reply]
@jamessimone As you might have already seen from the Exception Emails, we are getting the error again on our Org. There haven't been any changes made to the Rollup Metadata or the Apex Rollup Version. So I am still clueless as to what may cause this, but I can say the Error doesn't seem to be fixed yet :)

**Reply by jamessimone** [reply]
@fabm97 good to know! And this is on the latest version?

**Reply by fabm97** [reply]
@jamessimone Oh no, we are still on 1.4.7, I will update to the newest version and let you know if the error persists :)

**Reply by fabm97** [reply]
@jamessimone It seems the Error is fixed with version 1.5.3 :) Thanks for your help! 
However we now seem to have a new issue:
`Failed to process Queueable job for class RollupAsyncProcessor for job ID 7071v0000Az7nyY.

caused by: System.SObjectException: SObject row was retrieved via SOQL without querying the requested field: Meter2GenerationUnitRelationship__c.AdjustedCCId__c

Class.RollupCalculator.StringRollupCalculator.handleConcat: line 870, column 1
Class.RollupCalculator.performRollup: line 210, column 1
Class.RollupAsyncProcessor.getCalculator: line 1168, column 1
Class.RollupAsyncProcessor.getUpdatedLookupItemsByRollup: line 1048, column 1
Class.RollupAsyncProcessor.process: line 611, column 1
Class.RollupAsyncProcessor.QueueableProcessor.execute: line 455, column 1`

The metadata Type looks like this:
![image](https://user-images.githubusercontent.com/40338881/167875211-5d86504e-2285-4ddd-9fca-af4247d16191.png)
It is started from an Apex Trigger on the Meter2GenerationUnit Object

**Reply by jamessimone** [reply]
@fabm97 I've created #312 to track this issue for you!
